code/modules/cargo/exports.dm

```diff
@@ -183,6 +197,51 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
 
 GLOBAL_LIST_EMPTY(exports_list)
 
+/**
+  * Calculates the exact export value of the object based on the object's datum materials.
+  *
+  * loops through the atom's material datums and then multiplies that amount by it's relative worth, and sums the total.
+  */
+/datum/export/proc/get_material_cost(atom/movable/A)
+	for(var/i in A.custom_materials)
+		var/datum/material/M = i
+		. += M.value_per_unit * A.custom_materials[M]
+	. = round(.)  //floor of the value
+
+/**
+  * Sells an object using it's material cost.
+  *
+  * Works similarly to the sell object proc, but utilizes the get_material_cost proc in order to obtain it's raw value in terms of materials.
+  * Similarly checks for pricetags to split export profit the same way.
+  */
+/datum/export/proc/sell_material_item(obj/O, datum/export_report/report, dry_run = TRUE, apply_elastic = TRUE)
+	///This is the value of the object, as derived from material datums.
+	var/material_cost = get_material_cost(O)
+	///Quantity of the object in question.
+	var/amount = get_amount(O)
+	///Utilized in the pricetag component. Splits the object's profit when it has a pricetag by the specified amount.
+	var/profit_ratio = 0
+
+	if(amount <=0 || material_cost <=0)
+		return FALSE
+	unit_name = "[O.name]"
+	profit_ratio = SEND_SIGNAL(O, COMSIG_ITEM_SPLIT_PROFIT_MATERIAL)
+	material_cost = material_cost * ((100 - profit_ratio) * 0.01)
+	if(dry_run == FALSE & COMSIG_ITEM_SPLIT_VALUE)
```

Comment:
Could you explain what this check is doing?

Reply:
So, Dry run is called as true when using an item like an export scanner to find the export value of an object. Here I'm checking to make sure that the object is actually being sold by having dry_run as false, and that there exists on a COMSIG_ITEM_SPLIT_VALUE for the object, as the following line sends the actual signal that gives part of the sold item's profit to the person who used a price tag on it.

Reply:
Here you can see the operator precedence in BYOND:
http://www.byond.com/docs/ref/info.html#/operator
If `dry_run` is a true/false boolean then this will resolve itself in either `1 & COMSIG_ITEM_SPLIT_VALUE` or `0 & COMSIG_ITEM_SPLIT_VALUE`
Since `COMSIG_ITEM_SPLIT_VALUE` is a constant then the operation is superfluous.

I'm not sure what you mean with "and that there exists on a COMSIG_ITEM_SPLIT_VALUE for the object", but it's not done this way.